fasthx/main.py

```diff
@@ -129,17 +134,19 @@ class Jinja:
     def __call__(
         self, template_name: str, *, no_data: bool = False
     ) -> Callable[[Callable[_P, Any | Awaitable[Any]]], Callable[_P, Awaitable[Any | Response]]]:
-        """
-        Decorator for rendering a route's return value to HTML using the Jinja2 template
+        """Decorator for rendering a route's return value to HTML using the Jinja2 template
         with the given name.
 
         Arguments:
+        ---------
             template_name: The name of the Jinja2 template to use.
             no_data: If set, the route will only accept HTMX requests.
-        """
 
-        def render(result: Any, *, context: dict[str, Any], request: Request) -> HTMLResponse:
-            return self.templates.TemplateResponse(name=template_name, request=request, context=result)
+        """  # noqa: D205, D401
+
+        def render(result: Any, *, context: dict[str, Any], request: Request): # type: ignore[no-untyped-def]
+            context_with_request = {"request": request, **context}
```

Comment:
It would be nice to move this logic into a `_build_context(result: Any, *, route_context: dict[str, Any], request: Request) -> dict[str, Any]` hook method that can be overridden by people who would also like to include the route's "context" (its resolved dependencies) in the Jinja rendering context. See issue #4

Reply:
Also, I think you're making a mistake here, see the docs of `HTMXRenderer`: `context` is the route's context (=its resolved dependencies) in this setting, not the Jinja rendering context. Actually, the route's return value will end up being the Jinja rendering context. Again, please have a look at #4

Reply:
Thanks for explaining. I admit I didn't see #4, feel free to close this one.